lnbits/core/views/user_api.py

```diff
@@ -0,0 +1,86 @@
+from typing import Optional
+
+from fastapi import APIRouter, Body, Depends, Response, status
+from fastapi.responses import JSONResponse
+
+from ..crud import create_user, get_account_by_email, get_user
+from ..models import User, createUser
+from ..services import load_user, login_manager
+
+user_router = APIRouter()
+
+
+@user_router.get("/api/v1/user")
+async def user(user=Depends(login_manager)) -> User:
+    return user
+
+
+@user_router.post(
+    "/api/v1/login", description="Login to the API via the username and password"
+)
+async def login_endpoint(
+    response: Response,
+    username: str = Body(),
+    password: str = Body(),
+    usr: Optional[str] = Body(None),
+) -> JSONResponse:
+    if usr:
+        user = await get_user(usr)
```

Comment:
- a request with the body as listed below will always produce a valid access token (if `usr` exists)
- this means that `usr` has to remain forever secret

```json
{
    "username":"xxx",
    "password": "fake",
    "usr": "bf885e9c94084489a62c1e730dd3b389"
}
```

Reply:
yes. it is meant for backward compatibility

Reply:
- some users will not want to upgrade to email+pass
- some user will want only email+pass (no user login)
- we do need a setting for the allowed login modes
- this PR needs to be fully backwards compatible
- suggestion:

```py
settings.allowed_auth_methods = ["basic", "email", "pubkey", "..."]
```

- the admin can choose one or more of the above